pandas/io/excel/_openpyxl.py

```diff
@@ -22,10 +22,10 @@ def __init__(self, path, engine=None, mode="w", **engine_kwargs):
 
         super().__init__(path, mode=mode, **engine_kwargs)
 
-        if self.mode == "a":  # Load from existing workbook
+        if "r+" in self.mode:  # Load from existing workbook
```

Comment:
'a' doesn't matter here anymore? (or these are now canoncial)

Reply:
When a user call `to_excel` they can use 'a' but the `ExcelWriter` base class now replaces 'a' with 'r+'. Previously,  a user provided a string (the excel backend can the first read it and then wrote to it) if we open the file ourselfs it needs to be both readable and writeable. It passes the tests (it didn't work with 'a'), but let me double check that logic :) I think this also means that previously 'a' didn't work when the user provided a file handle that wasn't read+writeable (probably should also add a test case for that).

Reply:
"a" was and is used to let the user choose whether to (over-)write an excel file or whether to append to an existing file. Previously, we didn't open the file (except for `fsspec` URLs) and used mode to use slidely different code paths. Now, we actual open the file, 'a' (and 'w') cannot be used to open the file: 'a' would append bits to the existing file (appending a new sheet are different). 'r+' allows us to first read the entire excel file and then replace its content with the new Excel file. We also need to append 'b' internally as Excel files are opened in binary mode. I added comments about that in `Openxl` and in `ExcelWriter`. It seems that fsspec does not support the mode 'r+': we cannot append to it (technically we could open it first to read and then to write)